### PR TITLE
Check subdomain on Create/Update

### PR DESCRIPTION
**Pull request by dwasinge**
This PR should replace https://github.com/rht-labs/lodestar-backend/pull/116.

Same changes as 116 with the following modifications:
- changed the mongo query to use a case insensitive search
- using WebApplicationException instead of the removed Exception
- put subdomain validation into their own methods.

### CODE REVIEW

File: src/main/java/com/redhat/labs/lodestar/service/EngagementService.java [hunk 1]
```diff
@@ -199,6 +201,65 @@ void validateCustomerAndProjectNames(Engagement toUpdate, Engagement existing) {
...
+    }
+
+    /**
+     * Return false if subdomain is null, empty, or is not found in the data store.  Otherwise, true.
+     * 
+     * @param subdomain
+     */
+    boolean doesSubdomainExist(String subdomain) {
+
+        if (subdomain == null || subdomain.isBlank() || getBySubdomain(subdomain).isEmpty()) {
```
Comment (mcanoy) [inline src/main/java/com/redhat/labs/lodestar/service/EngagementService.java:245]: not a big fan of `return true` how do you feel about removing the if statement and just have a `return (equality test)`

File: src/main/java/com/redhat/labs/lodestar/resource/EngagementResource.java [hunk 2]
```diff
@@ -264,6 +266,19 @@ public Engagement put(@PathParam("id") String uuid, @Valid Engagement engagement
...
     }
 
+    @HEAD
+    @SecurityRequirement(name = "jwt", scopes = {})
+    @Path("/subdomain/{subdomain}")
+    @APIResponses(value = {
+            @APIResponse(responseCode = "401", description = "Missing or Invalid JWT"),
+            @APIResponse(responseCode = "409", description = "Subdomain is already in use"),
+            @APIResponse(responseCode = "200", description = "Engagement resource found and metadata returned in headers") })
+    public Response uniqueSubdomain(@PathParam("subdomain") String subdomain) {
```
Comment (mcanoy) [inline src/main/java/com/redhat/labs/lodestar/resource/EngagementResource.java:276]: should probably use SC_OK if using SC_CONFLICT

**Review by mcanoy — COMMENTED** [review]
a couple of minor thoughts inline

File: src/main/java/com/redhat/labs/lodestar/resource/EngagementResource.java (hunk 2 above)
Comment (dwasinge) [reply src/main/java/com/redhat/labs/lodestar/resource/EngagementResource.java:276]: updated to use SC_OK

**Review by mcanoy — APPROVED** [review]

File: src/main/java/com/redhat/labs/lodestar/service/EngagementService.java (hunk 1 above)
Comment (dwasinge) [reply src/main/java/com/redhat/labs/lodestar/service/EngagementService.java]: updated these two methods as there were no null/blank checks on the direct api call.  used the suggested equality test in stead of if statement.

**Pull request merged**